src/mawpy/steps/incremental_clustering.py

```diff
@@ -288,8 +284,7 @@ def _get_locations_to_cluster_center_map(clusters_list):
     return locations_to_cluster_center_map
 
 
-def _run_for_user(df_by_user, spat_constr, dur_constr=None):
-
+def _run_for_user(df_by_user: pd.DataFrame, spat_constr: float, dur_constr: Optional[float] = None) -> pd.DataFrame:
```

Comment:
```suggestion
def _run_for_user(df_by_user: pd.DataFrame, spat_constr: float, dur_constr: float | None = None) -> pd.DataFrame:
```

The new way of doing Optional is actually just do `type | None`.

Reply:
updated.